cdc/split.cc

```diff
@@ -122,16 +109,20 @@ extract_row_updates(const row& r, column_kind ckind, const schema& schema) {
                         v.timestamp(),
                         v.is_live_and_has_ttl() ? v.ttl() : gc_clock::duration(0)
                     );
-                auto& updates = result[timestamp_and_ttl].nonatomic_updates;
+                auto& updates = result[timestamp_and_ttl].nonatomic_entries;
                 if (updates.empty() || updates.back().id != id) {
                     updates.push_back({id, {}});
                 }
                 updates.back().cells.push_back({std::move(k), std::move(v)});
             }
 
             if (desc.tomb) {
-                auto timestamp_and_ttl = std::pair(desc.tomb.timestamp, gc_clock::duration(0));
-                result[timestamp_and_ttl].nonatomic_deletions.push_back({id, desc.tomb});
+                auto timestamp_and_ttl = std::pair(desc.tomb.timestamp + 1, gc_clock::duration(0));
```

Comment:
If not using the timestamp of the actual collection mutations, why not at least add some (debug?) code that verifies that the respective timestamps are T and T+1?

Reply:
There is no such assumption.
Inside a `collection_mutation` there can be a tombstone with timestamp `T` and a cell with timestamp e.g. `T+123`. The user can create any kind of weird mutations with `begin batch ... apply batch`.